x-pack/plugins/task_manager/server/MONITORING.md

```diff
@@ -0,0 +1,242 @@
+Each section has a `timestamp` and a `status` which indicates when the last update to this setion took place and whether the health of this section was evaluated as `OK`, `Warning` or `Error`.
+
+The root has its own `status` which indicate the state of the system overall as infered from the `status` of the section.
+An `Error` status in any section will cause the whole system to display as `Error`.
+A `Warning` status in any section will cause the whole system to display as `Warning`.
+An `OK` status will only be displayed when all sections are marked as `OK`.
+
+The root `timestamp` is the time in which the summary was exposed (either to the DEBUG logger or the http api) and the `lastUpdate` is the last time any one of the sections was updated.
+
+#### The Configuration Section
+The `configuration` section summarizes Task Manager's current configuration, including dynamic configurations which change over time, such as `poll_interval` and `max_workers` which adjust in reaction to changing load on the system.
+
+These are "Hot" stats which are updated whenever a change happens in the configuration.
+
+#### The Workload Section
+The `workload` which summarizes the work load in the current deployment, listing the tasks in the system, their types and what their current status is.
+
+It includes three sub sections:
+  - The number of tasks scheduled in the system, broken down by type and status.
+  - The number of idle `overdue` tasks, whose `runAt` has expired.
+  - Execution density in the next minute or so (configurable), which shows how many tasks are scheduled to execute in the scope of each polling interval. This can give us an idea of how much load there is on the current Kibana deployment.
+
+These are "Cold" stat which are updated at a regular cadence, configured by the `monitored_aggregated_stats_refresh_rate` config.
+
+#### The Runtime Section
+The `runtime` tracks Task Manager's performance as it runs, making note of task execution time, _drift_ etc.
+These include:
+  - The time it takes a task to run (mean and median, using a configurable running average window, `50` by default)
+  - The average _drift_ that tasks experience (mean and median, using the same configurable running average window as above). Drift tells us how long after a task's scheduled a task typically executes.
+  - The polling rate (the timestamp of the last time a polling cycle completed) and the result [`No tasks | Filled task pool | Unexpectedly ran out of workers`] frequency the past 50 polling cycles (using the same window size as the one used for running averages)
+  - The `Success | Retry | Failure ratio` by task type. This is different than the workload stats which tell you what's in the queue, but ca't keep track of retries and of non recurring tasks as they're wiped off the index when completed.
+
+These are "Hot" stats which are updated reactively as Tasks are executed and interacted with.
+
+### Example Stats
```

Comment:
So I'm curious (again, without looking at the code yet!) how much of the `runtime` section is purely from a Kibana's internal bookkeeping.  And then how well this will work out in a multi-Kibana story.  It seems like trying to aggregate over all the Kibana instances is going to be hard for this section, so I can understand if the first iteration of this just returns the current Kibana instance's stats.  I think we should include the kibana server uuid value in here, in that case.  At least a human could just keep doing `curl`'s till you know you hit every server.

Reply:
> I think we should include the kibana server uuid

Yeah, I actually intended on doing that and forgot :)
I had thought of this for that exact reason.

Reply:
> how much of the runtime section is purely from a Kibana's internal bookkeeping

This isn't about `1s` vs. `2s`, it's about having visibility into what's going on in the server that might explain why a customer is seeing a certain outcome we didn't expect.
If, for example, average duration is `30s` on certain task type, but it's average _drift_ is `1m` on a `5m` schedule, then we know the issue is in TM itself, not in the task. We then know to look at things like, scheduleDensity`, `max_workers` etc.
This is all about getting indicators that help us advise customers.

Long term, we also want to shift the Health Status based on these things, but for now, in order to reach GA, this is all about getting some observability into that black box.